src/parent/reducers/reducers.js

```diff
@@ -8,6 +8,43 @@ function parentReducer(state = {}, action) {
     case ACTION_TYPES.CLOSE: {
         const newState = Object.assign({}, state);
         delete newState[action.windowName];
+
+        if (!Object.keys(newState).length) {
+            fin.desktop.Window.getCurrent().contentWindow.close();
+        }
+
+        return newState;
+    }
+
+    case ACTION_TYPES.DRAG_OUT: {
+        const newState = Object.assign({}, state);
+        newState.dragOut = {
+            stockCode: action.code,
+            stockName: action.name
+        };
+        console.log(newState);
+        return newState;
+    }
+
+    case ACTION_TYPES.DRAG_ACCEPT: {
```

Comment:
I think making use of reducer composition would clean this up - there could be a `dragAndDropReducer` to handle these actions. I think changing the state shape would help with this too. I think something like the example would work well - so reducer composition can be used and hopefully would solve some of the more awkward operations in these reducers - such as deleting the dragOut state and counting the number of child windows based on the number of keys in the state.

Just ideas though - might be easier to discuss :smiley: 

For example:

**Current**
```js
{
    window0001: childReducer,
    window0002: childReducer,
    dragOut: {}  //parent reducer handles this too
}
```

**Proposed**
```js
{
    childWindows: {
        window0001: childReducer,
        window0002: childReducer
    },
    dragOut: dragAndDropReducer
}
```